src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -0,0 +1,209 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    /// <summary>Provides an abstraction for time.</summary>
+    public abstract class TimeProvider
+    {
+        private readonly double _tickFrequency;
```

Comment:
This is no longer the "frequency" but rather "TimeSpan.TicksPerSecond / Frequency".